workspaces/jenkins/plugins/jenkins-backend/src/service/jenkinsInfoProvider.ts

```diff
@@ -291,4 +317,24 @@ export class DefaultJenkinsInfoProvider implements JenkinsInfoProvider {
       ]
     );
   }
+
+  private static getEntityOverrideURL(entity: Entity) {
+    return entity.metadata.annotations?.[
+      DefaultJenkinsInfoProvider.JENKINS_OVERRIDE_URL
+    ];
+  }
+
+  private static verifyUrlMatchesRegex(url: string, regexList: string[]) {
+    for (const regexString of regexList) {
+      try {
+        const regex = new RegExp(regexString);
+        if (regex.test(url)) {
+          return true;
+        }
+      } catch (e) {
+        throw new Error(`Invalid regex: ${regexString}`);
```

Comment:
This is a bit bad as it's checked on runtime and there's no way to validate this before first component with annotation is added. I would instead just add error log here and return false, WDYT?

Reply:
Good call @drodil, i changed it to log out the error but not return false
As one bad regex should not block out the checking of the other regex, I have added a test for this scenario fyi